addons/mail/static/src/models/thread/thread.js

```diff
@@ -1385,8 +1388,10 @@ function factory(dependencies) {
             if (this.channel_type === 'chat' && this.correspondent) {
                 return this.custom_channel_name || this.correspondent.nameOrDisplayName;
             }
-            if (this.channel_type === 'group') {
-                return this.name || this.members.map(partner => partner.nameOrDisplayName).join(', ');
+            if (this.channel_type === 'group' && !this.name) {
+                const partnerNames = this.members.map(partner => partner.nameOrDisplayName).sort();
+                const guestNames = this.guestMembers.map(guest => guest.name).sort();
+                return [...partnerNames, ...guestNames].join(", ");
```

Comment:
We might want to translate the `,`, as if I remember well some languages don't use that punctuation?

Reply:
In Japanese at least, the comma character is different (`、`)

Reply:
The `Intl.ListFormat` object exists for this purpose. However, I'm not sure the separator here should depend on the user's locale, since the names of the channel members themselves are not translatable. It would feel odd to join words written in the Latin alphabet using an ideographic comma, wouldn't it?

I think the translation of the separator should be contextual to the language of the list elements rather than to the user's locale, but this is not a problem we have a solution for at the moment, so I would suggest leaving things as it.

Reply:
It's not an easy solution to solve all problems. I think the best approach with the lesser effort would be to still make `", "` translatable. In most locales, it's unchanged and just `", "`. For Japanese work-environment, it just looks a bit weird to display (essentially Japanese) names with standard comma.

I think this really only affects Japanese, if I'm not mistaken it's the only language that has its own character for comma. This affects a very specific demography, so it's not a super big deal, especially when plain comma works (despite not "clean").

So my reasoning is just not over-engineering because it's very specific. But also it doesn't cost much to just wrap in `_t()` and make it cleaner for users that have Japanese locale.